21_node_management/22_search_nodes.txt

```diff
@@ -10,25 +10,154 @@ about the Nodes which have been already validated, and are managed by Rudder.
 
 ==== Quick Search
 
-The easiest search tool is the Quick search: type in the search field the first
-letters of the 'Rudder ID', 'Reference', or 'Hostname'; choose the accurate Node in
-the autocompletion list; validate and look at the Node information. This search
-tool can be very useful to help you create a new search in the Advanced Search.
-
-.Quick search the Node called +debian-node+
+You might have noticed the small text area at the top of the Rudder interface:
+it is the Quick Search bar. Its purpose is to enable a user to easily search for
+Rudder elements (Nodes, Groups, Directives, Parameters, Rules)  based on their
+name, id, description, inventory...
+
+image::./images/rudder-quicksearch.png[Quicksearch]
+
+An autocompletion list will appear as soon as Rudder detects an element it can
+identify, you just have to click on it to be redirected to the element
+configuration page.
+
+More complex search queries can be input using the "in:" and "is:" keywords,
+"is" targets Rudder objects by type, and "in" targets elements like name,
+description...
+
+Those keywords are used to refine a research in case a search query returns
+too many results.
+
+The available search keywords are:
+
+.is: keywords
+[frame="topbot",options="header"]
+|======================
+|keyword |Description
+|node      |Nodes
+|group     |Groups
+|parameter |Parameters
+|directive |Directives
+|rule      |Rules
+|======================
+
+.in: keywords (common)
+[frame="topbot",options="header"]
+|======================
+|keyword |Search for
+|name |Names
+|id |IDs
+|description, long_description |Descriptions
+|isenabled |Enabled elements
+|======================
+
+.in: keywords (nodes)
+[frame="topbot",options="header"]
+|======================
+|keyword |Search for
+|hostname |Hostnames
+|os_type |OS types (windows, linux, aix...)
+|os_name |OS Names (Windows Server 2008, Debian... )
+|os_version |OS versions (8, 2008 R2, ...)
+|os |OS Full Names (Debian GNU/Linux 6.0.10 (squeeze)...)
+|os_kernel_version |OS Kernel versions (3.16, 5.1...)
+|os_service_pack |OS Service Packs (for Windows and SuSE Linux)
+|architecture |OS Architectures (amd64, x86_64, i386)
+|ram |Machine memory
+|ips |Network IP addresses
+|policy_server_id |ID of a node's policy server (root...)
+|properties |Node properties (arbitrary key=values associated to a node)
+|rudder_roles |Rudder roles (rudder-reports, rudder-ldap...)
+|======================
+
+.in: keywords (groups)
+[frame="topbot",options="header"]
+|======================
+|keyword |Search for
+|dynamic |Dynamic groups
+|======================
+
+.in: keywords (directives)
+[frame="topbot",options="header"]
+|======================
+|keyword |Search for
+|dir_param_name |Directive parameter names (Enforce file content...)
```

Comment:
It would be the name of the parameter in the Technique metadatra.xl : like GENERIC_FILE_CONTENT_PATH for path of file in "Enforce a file content" technique